io/cmd.go

```diff
@@ -94,9 +94,11 @@ func RunCmdWithOutputParser(config CmdConfig, prompt bool, regExpStruct ...*CmdO
 			for _, regExp := range regExpStruct {
 				matched := regExp.RegExp.Match([]byte(line))
 				if matched {
-					regExp.MatchedResults = regExp.RegExp.FindStringSubmatch(line)
-					regExp.Line = line
-					line, err = regExp.ExecFunc(regExp)
+					results := CmdOutputPattern{
+						MatchedResults: regExp.RegExp.FindStringSubmatch(line),
+						Line:           line,
```

Comment:
```suggestion
						RegExp:         regExp.RegExp,
						MatchedResults: regExp.RegExp.FindStringSubmatch(line),
						Line:           line,
						ExecFunc:       regExp.ExecFunc,
```

Even better if you could export this logic to a new function and reuse it